arangod/Agency/Supervision.cpp

```diff
@@ -3554,6 +3554,12 @@ void Supervision::checkUndoLeaderChangeActions() {
         return Result{TRI_ERROR_BAD_PARAMETER, "collection missing"};
       }
 
+      auto maybeShardID = ShardID::shardIdFromString(id);
+      if (ADB_UNLIKELY(maybeShardID.fail())) {
+        return Result{TRI_ERROR_BAD_PARAMETER,
```

Comment:
It would probably be better to return the error message generated by `shardIdFromString()` here. If it returns a `ResultT` and there is an error, that method should have populated the error message with something useful already, so we can simply return it here.

Reply:
Fixed with 134fb22ab67